docs/architecture/adr-071-crypto-v2-multi-curve.md

```diff
@@ -0,0 +1,239 @@
+
+The new `Keyring` interface will serve as a central hub for managing and fetching `CryptoProviders`. To ensure a smoother migration path, the new Keyring will be backward compatible with the previous version. Since this will be the main API from which applications will obtain their `CryptoProvider` instances, the proposal is to extend the Keyring interface to include the methods:
+
+
+```go
+type KeyringV2 interface {
+  // methods from Keyring/v1
+  
+  // ListCryptoProviders returns a list of all the stored CryptoProvider metadata.
+  ListCryptoProviders() ([]ProviderMetadata, error)
+  
+  // GetCryptoProvider retrieves a specific CryptoProvider by its id.
+  GetCryptoProvider(id string) (CryptoProvider, error)
+}
+```
+
+*Note*: Methods to obtain a provider from a public key or other means that make it easier to load the desired provider can be added.
+
+##### Especial use case: remote signers
+
+It's important to note that the `CryptoProvider` interface is versatile enough to be implemented as a remote signer. This capability allows for the integration of remote cryptographic operations, which can be particularly useful in distributed or cloud-based environments where local cryptographic resources are limited or need to be managed centrally.
+
+
+## Alternatives
+
+It is important to note that all the code presented in this document is not in its final form and could be subject to changes at the time of implementation. The examples and implementations discussed should be interpreted as alternatives, providing a conceptual framework rather than definitive solutions. This flexibility allows for adjustments based on further insights, technical evaluations, or changing requirements as development progresses.
+
+## Decision
+
+We will:
+
+* Leverage crypto providers
+* Refactor the module structure as described above.
+* Define types and interfaces as the code attached.
+* Refactor existing code into new structure and interfaces.
+* Implement Unit Tests to ensure no backward compatibility issues.
+
+## Consequences
+
+### Impact on the SDK codebase
+
+We can divide the impact of this ADR into two main categories: state machine code and client related code.
+
+#### Client
+
+The major impact will be on the client side, where the current `Keyring` interface will be replaced by the new `KeyringV2` interface. At first, the impact will be low since `CryptoProvider` is an optional field in the `Record` message, so there's no mandatory requirement for migrating to this new concept right away. This allows a progressive transition where the risks of breaking changes or regressions are minimized.
+
+
+#### State Machine
+
+The impact on the state machine code will be minimal, the modules affected (at the time of writing this ADR)
+are the `x/accounts` module, specifically the `Authenticate` function and the `x/auth/ante` module. This function will need to be adapted to use a `CryptoProvider` service to make use of the `Verifier` instance.
+
```

Comment:
i dont think we should do this, this makes devux slightly worse since now they have to implement a cryptoprovider to make things work instead of simply verifying the signature right away.

Reply:
Yep, giving a second thought this is not mandatory